sdk/build-scripts/src/utils/buildPackage.ts

```diff
@@ -0,0 +1,219 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { promises as fs, existsSync } from 'fs';
+import * as path from 'path';
+import { BuildOptions, build } from 'esbuild';
+import { execSync } from 'child_process';
+
+interface PackageJSON {
+	name?: string;
+	type?: 'module' | 'commonjs';
+	exports?: Record<string, string | Record<string, string>>;
+	files?: string[];
+	types?: string;
+	import?: string;
+	main?: string;
+	private?: boolean;
+	sideEffects?: boolean;
+}
+
+const ignorePatterns = [/\.test.ts$/];
+
+export async function buildPackage(buildOptions?: BuildOptions) {
```

Comment:
the only difference between this and the original file is that I added this `buildOptions` arg